Source/carthage/BuildVersion.swift

```diff
@@ -5,8 +5,43 @@ import ReactiveTask
 import Result
 import Tentacle
 
+struct ExpirableVersion: Codable {
+	let version: SemanticVersion?
+	let expiredDate: Date
+
+	init(version: SemanticVersion?, shelfLife: TimeInterval) {
+		self.version = version
+		self.expiredDate = Date(timeIntervalSinceNow: shelfLife)
+	}
+
+	init?(fromJSONData data: Data?) {
+		guard let wrappedData = data, let wrapped = try? JSONDecoder().decode(ExpirableVersion.self, from: wrappedData) else {
+			return nil
+		}
+		self = wrapped
+	}
+
+	func encodedJSONData() -> Data? {
+		return try? JSONEncoder().encode(self)
+	}
+
+	var isExpired: Bool {
+		return Date().compare(expiredDate) == ComparisonResult.orderedDescending
+	}
+}
+
 /// The latest online version as a SemanticVersion object.
 public func remoteVersion() -> SemanticVersion? {
+	let userDefaults = UserDefaults.standard
+	let cachedKey = "latestCarthageRemoteVersion"
+
+	let expirableVersionData = userDefaults.data(forKey: cachedKey)
+	let expirableVersion = ExpirableVersion(fromJSONData: expirableVersionData)
+
+	guard expirableVersion == nil || expirableVersion!.isExpired else {
+		return expirableVersion!.version
+	}
```

Comment:
Instead of adding this check to this function, can you create a new function that checks and then calls this one?